src/rosdistro/manifest_provider/github.py

```diff
@@ -39,32 +39,24 @@
     from urllib2 import URLError
 
 from rosdistro import logger
-from rosdistro.manifest_provider import get_release_tag
-from rosdistro.manifest_provider.git import check_remote_tag_exists
 
 
 def github_manifest_provider(_dist_name, repo, pkg_name):
     assert repo.version
-    if 'github.com' not in repo.url:
+    server, path = repo.get_url_parts()
+    if server != 'github.com':
         logger.debug('Skip non-github url "%s"' % repo.url)
         raise RuntimeError('can not handle non github urls')
 
-    release_tag = get_release_tag(repo, pkg_name)
+    release_tag = repo.get_release_tag(pkg_name)
 
-    if not check_remote_tag_exists(repo.url, release_tag):
+    if release_tag not in repo.remote_tags:
```

Comment:
Yes, I think it would be good to add a `has_remote_tag` method and use it here. The current repo impl. can still continue to get all tags and implement that method by only considering the already fetch data.

`remote_tags` would just not be public API which needs to be implemented by alternative impl. It allows an alternative implementation to check the requested tag on demand.